model/inline/src/main/scala/aqua/model/inline/raw/ApplyIntoCopyRawInliner.scala

```diff
@@ -0,0 +1,59 @@
+package aqua.model.inline.raw
+
+import aqua.model.{CallModel, CallServiceModel, LiteralModel, OpModel, SeqModel, ValueModel, VarModel}
+import aqua.model.inline.{Inline, SeqMode}
+import aqua.model.inline.MakeStructRawInliner.createObj
+import aqua.model.inline.RawValueInliner.unfold
+import aqua.model.inline.state.{Arrows, Exports, Mangler}
+import aqua.raw.value.{IntoCopyRaw, LiteralRaw}
+import aqua.types.ScalarType
+import cats.data.{Chain, NonEmptyMap, State}
+import scribe.Logging
+import cats.syntax.traverse.*
+import cats.syntax.monoid.*
+import cats.syntax.functor.*
+import cats.syntax.flatMap.*
+import cats.syntax.apply.*
+
+object ApplyIntoCopyRawInliner extends Logging {
+
+  private def copyObj(value: VarModel, fields: NonEmptyMap[String, ValueModel], result: VarModel): OpModel.Tree = {
+    val args = fields.toSortedMap.toList.flatMap { case (name, value) =>
+      LiteralModel.fromRaw(LiteralRaw.quote(name)) :: value :: Nil
+    }
+    CallServiceModel(
+      LiteralModel("\"json\"", ScalarType.string),
+      "puts",
+      CallModel(
+        value +: args,
+        CallModel.Export(result.name, result.`type`) :: Nil
+      )
+    ).leaf
+  }
+
+  def apply[S: Mangler: Exports: Arrows](
+    value: VarModel, intoCopy: IntoCopyRaw
+  ): State[S, (VarModel, Inline)] = {
+    value match {
+      case v @ VarModel(name, _, _) =>
+        for {
+          name <- Mangler[S].findAndForbidName(name + "_obj_copy")
```

Comment:
- Pattern matching seems unnecessary there
- Variable shadowing looks overcomplicated, not sure which `name` is used where